adal/util.py

```diff
@@ -55,7 +55,11 @@ def add_default_request_headers(self, options):
     headers[AdalIdParameters.SKU] = AdalIdParameters.PYTHON_SKU
     headers[AdalIdParameters.VERSION] = adal.__version__
     headers[AdalIdParameters.OS] = sys.platform
-    headers[AdalIdParameters.CPU] = 'x64' if platform.architecture()[0] == '64bit' else 'x86'
+
+    try:
+        headers[AdalIdParameters.CPU] = 'x64' if platform.architecture()[0] == '64bit' else 'x86'
+    except:
+        headers[AdalIdParameters.CPU] = 'unknown'

```

Comment:
This exception handling introduces a new value `unknown` in the header, which our service end may or may not expect that.

How about using a different approach which is supposed to work always?

    headers[AdalIdParameters.CPU] = 'x64' if sys.maxsize > 2**32 else 'x86'